dbt/adapters/snowflake/connections.py

```diff
@@ -447,41 +458,50 @@ def add_query(self, sql, auto_begin=True, bindings=None, abridge_sql_log=False):
             bindings = tuple(bindings)
 
         queries = self._split_queries(sql)
-
-        for individual_query in queries:
+        stripped_queries = []
+        for query in queries:
             # hack -- after the last ';', remove comments and don't run
             # empty queries. this avoids using exceptions as flow control,
             # and also allows us to return the status of the last cursor
             without_comments = re.sub(
                 re.compile(r"(\".*?\"|\'.*?\')|(/\*.*?\*/|--[^\r\n]*$)", re.MULTILINE),
                 "",
-                individual_query,
+                query,
             ).strip()
 
-            if without_comments == "":
-                continue
+            if without_comments != "":
+                stripped_queries.append(query)
+
+        if set(query.lower() for query in stripped_queries).issubset({"begin;", "commit;"}):
+            # if all we get is `begin;` and/or `commit;`
+            # raise a warning, then run as standard queries to avoid an error downstream
+            message = (
+                "Explicit transactional logic should be used only to wrap "
+                "DML logic (MERGE, DELETE, UPDATE, etc). The keywords BEGIN; and COMMIT; should "
+                "be placed directly before and after your DML statement, rather than in separate "
+                "statement calls or run_query() macros."
+            )
+            logger.warning(line_wrap_message(warning_tag(message)))
+            for query in stripped_queries:
+                connection, cursor = self.add_standard_query(
+                    query, auto_begin, bindings, abridge_sql_log
+                )
```

Comment:
I think we need a `return` here, right? If we get a set queries that are only `begin;` and/or `commit;`, then we want the this condition and that's all. In other words, we don't want the for loop below to run and execute `super().add_begin_query()` and `super().add_commit_query()`. I might not be remembering the requirements correctly.